### PR TITLE
FAQ & API Gateway doc

### PR DESCRIPTION
**Pull request by nie7321**
## Overview

This adds a doc about API Gateway & how we recommend using it.

I also added some TF FAQ items.

Looking for ideas for more!

## Checklist

- [x] Proof-reading
- [x] Any diagrams/figures have their source files included

### CODE REVIEW

File: infrastructure/api-gateway.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+The AWS [API Gateway](https://docs.aws.amazon.com/apigateway/) is Amazon's serverless platform for API development and delivery. It has many features found in [Apigee](https://cloud.google.com/apigee/), and can be used to similar purpose.
+
+:::tip HTTP APIs for API Gateway
+Amazon has a simplified and less-expensive version of API Gateway in beta, called [HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api.html). It drops support for most of the features we already rely on Apigee for.
+
+We have not deployed this service anywhere yet, but it sounds like a strong candidate to replace all API Gateway usage.
+:::
+
+## Usage
+Our typical use-case for API Gateway is as the load balancer for serverless APIs running on Lambda. We do not use most of the features it offers: documentation, API key management, rate limiting, and request/response re-writing are all handled in Apigee.
```
Comment (cecolby11) [inline infrastructure/api-gateway.md]: Should we add more discussion about when you should proxy your api gateway endpoint(s) through Apigee? Seems like a good topic for how we use API GW at NU. I'm not sure what the criteria is, I'm thinking if you only call the API Gateway endpoint internally in your application it wouldn't need to go through Apigee, but as soon as you want to call it from any other application it should go through Apigee, even if you own the other application.

Comment (bab4379) [reply infrastructure/api-gateway.md]: I'm torn on Apigee, I think the idea is "everything" goes through Apigee.  Not sure how I feel about the overhead (time and latency) of using Apigee for a service only your application would use.  But it does allow you to easily expose that service to other applications without rework.

Comment (nie7321) [reply infrastructure/api-gateway.md]: >  I think the idea is "everything" goes through Apigee.

Welllll, not *everything*. An SPA (RP, ECATS) is still going to need to talk to API Gateway directly, 'cuz cookies.

Comment (nie7321) [reply infrastructure/api-gateway.md]: I've added a subsection to "Usage" about interacting with Apigee (and the upcoming vanity name).

Comment (bab4379) [reply infrastructure/api-gateway.md]: I was not really thinking that through.  Not sure what the distinction should be, seems like next week the SPA cookie issue could be solved

**Pull request merged**